base/steer/FairRootManager.cxx

```diff
@@ -139,8 +194,109 @@ FairRootManager::~FairRootManager()
   LOG(DEBUG) << "Leave Destructor of FairRootManager" << FairLogger::endl;
   delete fEventHeader;
   delete fSourceChain;
+
+  // Global cleanup
+  --fgCounter;
+  if ( ! fgCounter ) {
+    delete fgIsFillLocks;
+    fgIsFillLocks = 0;
+  }
+
+  //
+#if ( ROOT_VERSION_CODE >= ROOT_VERSION(6,9,3) )
+  lk.unlock();
+#endif
+  if ( fDebugMT ) {
+    LOG(INFO) << "Released lock and done FairRootManager::~FairRootManager in "
+      << fId << " " << this << FairLogger::endl;
+  }
 }
+
+//
+// private methods for MT
+//
+
 //_____________________________________________________________________________
+void  FairRootManager::LogMessage(const TString& message)
+{
+/// Output one line log message followed with the instance Id and pointer
+
+  if ( fDebugMT ) {
+    LOG(INFO) << message << " in " << fId << " " << this << FairLogger::endl;
+  }
+}
+
+//_____________________________________________________________________________
+void  FairRootManager::FillWithTmpLock()
+{
+/// Fill the Root tree.
+
+  LogMessage("Going to lock for Fill");
```

Comment:
Can you please move the log message inside the `#if .. #endif`? There are multiple places where this needs to be applied. Thank you.